mip/highs.py

```diff
@@ -0,0 +1,1048 @@
+        n = self.num_cols()
+        num_col = ffi.new("int*")
+        costs = ffi.new("double[]", n)
+        lower = ffi.new("double[]", n)
+        upper = ffi.new("double[]", n)
+        num_nz = ffi.new("int*")
+        check(
+            self._lib.Highs_getColsByRange(
+                self._model,
+                0,  # from_col
+                n - 1,  # to_col
+                num_col,
+                costs,
+                lower,
+                upper,
+                num_nz,
+                ffi.NULL,  # matrix_start
+                ffi.NULL,  # matrix_index
+                ffi.NULL,  # matrix_value
+            )
+        )
+        obj_expr = mip.xsum(
+            costs[i] * self.model.vars[i] for i in range(n) if costs[i] != 0.0
+        )
+        obj_expr.add_const(self.get_objective_const())
+        obj_expr.sense = self.get_objective_sense()
+        return obj_expr
+
+    def get_objective_const(self: "SolverHighs") -> numbers.Real:
+        offset = ffi.new("double*")
+        check(self._lib.Highs_getObjectiveOffset(self._model, offset))
+        return offset[0]
+
+    def _all_cols_continuous(self: "SolverHighs"):
+        n = self.num_cols()
+        integrality = ffi.new(
+            "int[]", [self._lib.kHighsVarTypeContinuous for i in range(n)]
+        )
+        check(
+            self._lib.Highs_changeColsIntegralityByRange(
+                self._model, 0, n - 1, integrality
+            )
+        )
+
+    def _reset_var_types(self: "SolverHighs"):
+        var_type_map = {
+            mip.CONTINUOUS: self._lib.kHighsVarTypeContinuous,
+            mip.BINARY: self._lib.kHighsVarTypeInteger,
+            mip.INTEGER: self._lib.kHighsVarTypeInteger,
+        }
+        integrality = ffi.new("int[]", [var_type_map[vt] for vt in self._var_type])
+        n = self.num_cols()
+        check(
+            self._lib.Highs_changeColsIntegralityByRange(
+                self._model, 0, n - 1, integrality
+            )
+        )
+
+    def relax(self: "SolverHighs"):
+        # change integrality of all columns
+        self._all_cols_continuous()
+        self._var_type = [mip.CONTINUOUS] * len(self._var_type)
+
+    def generate_cuts(
+        self,
+        cut_types: Optional[List[mip.CutType]] = None,
+        depth: int = 0,
+        npass: int = 0,
+        max_cuts: int = mip.INT_MAX,
+        min_viol: numbers.Real = 1e-4,
+    ) -> "mip.CutPool":
+        raise NotImplementedError("HiGHS doesn't support manual cut generation.")
+
+    def clique_merge(self, constrs: Optional[List["mip.Constr"]] = None):
+        raise NotImplementedError("HiGHS doesn't support clique merging!")
+
+    def optimize(
+        self: "SolverHighs",
+        relax: bool = False,
+    ) -> "mip.OptimizationStatus":
+        if relax:
+            # Temporarily change variable types. Original types are still stored
+            # in self._var_type.
+            self._all_cols_continuous()
+
```

Comment:
@rschwarz Thank you very much for providing this PR!

As it seems, the maximum gap parameters do not reach the HiGHS solver:
```suggestion

        self.set_mip_gap(self.model.max_mip_gap)
        self.set_mip_gap_abs(self.model.max_mip_gap_abs)
        
```

Reply:
I haven't tested this, but I assumed that these calls would be made automatically, via https://github.com/coin-or/python-mip/blob/9945ed1879cedb990c43eba2b37893c85b308d19/mip/model.py#L1297 or so. Apparently this is not the case and your suggestion is consistent with the https://github.com/coin-or/python-mip/blob/9945ed1879cedb990c43eba2b37893c85b308d19/mip/cbc.py#L1213, so I'll happily adopt it.